packages/features/webhooks/pages/webhooks-view.tsx

```diff
@@ -1,74 +1,172 @@
+    name: string | null;
+    image?: string | undefined;
+    teamId: number | null | undefined;
+  }[];
+}) => {
   const { t, isLocaleReady } = useLocale();
+
+  if (!profiles || profiles.length < 2) {
+    return (
+      <Button
+        color="primary"
+        data-testid="new_webhook"
+        StartIcon={Plus}
+        href={`${WEBAPP_URL}/settings/developer/webhooks/new${!!teamId ? `?teamId=${teamId}` : ""}`}>
+        {isLocaleReady ? t("new") : <SkeletonText className="h-4 w-24" />}
+      </Button>
+    );
+  }
   return (
-    <Button
-      color="secondary"
-      data-testid="new_webhook"
-      StartIcon={Plus}
-      href={`${WEBAPP_URL}/settings/developer/webhooks/new`}>
-      {isLocaleReady ? t("new_webhook") : <SkeletonText className="h-4 w-24" />}
-    </Button>
+    <Dropdown>
+      <DropdownMenuTrigger asChild>
+        <Button color="primary" StartIcon={Plus}>
+          {isLocaleReady ? t("new") : <SkeletonText className="h-4 w-24" />}
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent sideOffset={14} align="end">
+        <DropdownMenuLabel>
+          <div className="text-xs">{t("create_for").toUpperCase()}</div>
+        </DropdownMenuLabel>
+        {profiles.map((profile, idx) => (
+          <DropdownMenuItem key={profile.slug}>
+            <DropdownItem
+              type="button"
+              StartIcon={(props) => (
+                <Avatar
+                  alt={profile.slug || ""}
+                  imageSrc={profile.image || `${WEBAPP_URL}/${profile.name}/avatar.png`}
+                  size="sm"
+                  {...props}
+                />
+              )}>
+              <Link href={`webhooks/new${profile.teamId ? `?teamId=${profile.teamId}` : ""}`}>
+                {profile.name}
+              </Link>
+            </DropdownItem>
+          </DropdownMenuItem>
+        ))}
+      </DropdownMenuContent>
+    </Dropdown>
   );
 };
 
-const WebhooksList = () => {
+const WebhooksList = ({ webhooksByViewer }: { webhooksByViewer: WebhooksByViewer }) => {
   const { t } = useLocale();
   const router = useRouter();
 
-  const { data: webhooks } = trpc.viewer.webhook.list.useQuery(undefined, {
-    suspense: true,
-    enabled: router.isReady,
-  });
+  const { profiles, webhookGroups } = webhooksByViewer;
+
+  const hasTeams = profiles && profiles.length > 1;
 
   return (
     <>
-      {webhooks?.length ? (
+      {webhookGroups && (
         <>
-          <div className="border-subtle mt-6 mb-8 rounded-md border">
-            {webhooks.map((webhook, index) => (
-              <WebhookListItem
-                key={webhook.id}
-                webhook={webhook}
-                lastItem={webhooks.length === index + 1}
-                onEditWebhook={() => router.push(`${WEBAPP_URL}/settings/developer/webhooks/${webhook.id} `)}
-              />
-            ))}
-          </div>
-          <NewWebhookButton />
+          {webhookGroups.length > 0 ? (
```

Comment:
```suggestion
          {webhookGroups.length && (
```
With big ternary conditionals I think its better if we do 2 explicits &&